### PR TITLE
Support child profilers adding their deviceProperties to logger

### PR DESCRIPTION
**Pull request by fwenguang**
This patch is intended to support our out-of-tree Kineto plugin in adding its deviceProperties.

### CODE REVIEW

**Comment by fwenguang** [comment]
Hi @sraikund16 , could you help review this?

**Comment by sraikund16** [comment]
This is a cool idea, thanks for submitting this. I think this will affect our internal testing so I'll try to dedicate some time to make sure things work this week.

File: libkineto/src/output_membuf.h [hunk 1]
```diff
@@ -66,6 +66,13 @@ class MemoryTraceLogger : public ActivityLogger {
     metadata_ = metadata;
   }
 
+  void handleTraceStart(
```
Comment (sraikund16) [inline libkineto/src/output_membuf.h:69]: Same as comment above

File: libkineto/src/CuptiActivityProfiler.cpp [hunk 2]
```diff
@@ -306,7 +308,17 @@ void CuptiActivityProfiler::processTraceInternal(ActivityLogger& logger) {
   for (auto& pair : versionMetadata_) {
     addMetadata(pair.first, pair.second);
   }
-  logger.handleTraceStart(metadata_);
+  std::vector<std::string> device_properties;
+  if (auto props = devicePropertiesJson(); !props.empty()) {
```
Comment (sraikund16) [inline libkineto/src/CuptiActivityProfiler.cpp:312]: nit: can we define these vars outside of the if statement? I think our internal linter will complain about lines like these

**Review by aaronenyeshi — APPROVED** [review]

**Pull request closed without merging**